src/analyzing/AST_to_IL.ml

```diff
@@ -1367,6 +1413,29 @@ and expr_stmt env (eorig : G.expr) tok : IL.stmt list =
       [ mk_s (Instr fake_i) ]
   | ss'' -> ss''
 
+and mk_class_construction env lval origin_exp ty cons_id_info args : stmt list =
+  (* x = new T(args) *)
+  (* HACK(new): Because of field-sensitivity hacks, we need to know to which
```

Comment:
This comment seems to have been copied, in the original place it makes sense and it should remain, but here we would need another comment. Just explain that we encode `x = new T(args)` as `x = new x.T(args)` so that taint analysis knows that the `this` when calling `T` is the variable `x`. It's kinda tacky but works for now.